hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/client/ECReplicationConfig.java

```diff
@@ -36,6 +43,29 @@ public ECReplicationConfig(int data, int parity) {
     this.parity = parity;
   }
 
+  public ECReplicationConfig(String string) {
+    final Pattern pattern = Pattern.compile("(\\d+)-(\\d+)");
+    final Matcher matcher = pattern.matcher(string);
```

Comment:
This is a question more from my own interest point of view, as I know this is not a performance critical piece of code.

These two variables are effectively constants, but they are defined at the method level. Do you know if Java is smart enough to realise these are constants and evaluate the expression only once? Compared to defining these as "private static final" at the class level, will this perform the same or worse?

I will try to answer this myself with a quick benchmark if I get some time.

Reply:
Good point. Let me move it to a static final constant.

Answering the question is slightly harder. Compilation time I couldn't see any optimization:


```
javap -v org.apache.hadoop.hdds.client.ECReplicationConfig

...

  public org.apache.hadoop.hdds.client.ECReplicationConfig(java.lang.String);
    descriptor: (Ljava/lang/String;)V
    flags: ACC_PUBLIC
    Code:
      stack=3, locals=4, args_size=2
         0: aload_0
         1: invokespecial #1                  // Method java/lang/Object."<init>":()V
         4: ldc           #4                  // String (\d+)-(\d+)
         6: invokestatic  #5                  // Method java/util/regex/Pattern.compile:(Ljava/lang/String;)Ljava/util/regex/Pattern;
         9: astore_2
```

The call is really compiled as is to the method.

Runtime optimization it's another question, but I don't think that it's possible easily as hostpost JVM would require understanding about the idempotency of `Pattern.compile` (second call on `Pattern.compile` returns with a value which is equivalent to the first one.)

--> looks better to move the compile to a static field...